### PR TITLE
Change save models and add resume training

### PR DESCRIPTION
**Pull request by melisandeteng**
Change save and load function of the networks to save optimizers and resume training.

### CODE REVIEW

**Comment by melisandeteng** [comment]
Note that maskdepth_generator.py is actually not relevant for this pull request, it's work in progress

File: utils.py [hunk 1]
```diff
@@ -259,3 +259,24 @@ def print_opts(flats):
...
+        last_model_name {str} -- last model name
+    """
+    if os.path.exists(dirname) is False:
+        return None
+    gen_models = [
+        os.path.join(dirname, f)
+        for f in os.listdir(dirname)
+        if os.path.isfile(os.path.join(dirname, f)) and key in f and ".pth" in f
+    ]
+    if gen_models is None:
```
Comment (vict0rsch) [inline utils.py:278]: gen_models can't be `None` but it can be empty

File: maskdepth_generator.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
+import torch
```
Comment (vict0rsch) [inline maskdepth_generator.py:1]: Is everything here so different that it can't fit in the other generator's code?

Comment (melisandeteng) [reply maskdepth_generator.py:1]: No actually adding this maskdepth_generator.py was not my intention for this pull request, plus it's not the one from the right branch.

Comment (vict0rsch) [reply maskdepth_generator.py:1]: ok so you'll add a commit to the PR?

Comment (melisandeteng) [reply maskdepth_generator.py:1]: I'll try and revert my commit for this file

File: utils.py (hunk 1 above)
Comment (melisandeteng) [reply utils.py:278]: Ah this was from the master, I hadn't noticed.

Comment (melisandeteng) [reply utils.py:278]: Actually this function is not used in the current way of resuming training. I will delete it.

Comment (melisandeteng) [reply utils.py]: I ended up keeping it, just in case I change the resume training function later

**Pull request merged**